### PR TITLE
Feature/kafka core

### PR DESCRIPTION
**Pull request by yassire**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,37 @@
+HELP.md
+target/
+!.mvn/wrapper/maven-wrapper.jar
```
Comment (iebouihi) [inline .gitignore:3]: I think we should include only file to ignore

File: .gitignore [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
+!**/src/main/**/target/
+!**/src/test/**/target/
```
Comment (iebouihi) [inline .gitignore:5]: Same here

File: kafka/kafka-core/pom.xml [hunk 3]
```diff
@@ -1 +1,49 @@
...
+            <artifactId>spring-kafka</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.retry</groupId>
+            <artifactId>spring-retry</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <scope>provided</scope>
```
Comment (iebouihi) [inline kafka/kafka-core/pom.xml:36]: Why this dependency is declared is provided ? It means that projects using this project as dependency will provide this lib ?

**Review by iebouihi — APPROVED** [review]

File: kafka/kafka-core/pom.xml (hunk 3 above)
Comment (yassire) [reply kafka/kafka-core/pom.xml:36]: This dependency is to generate data manipulation methods like getters and setters and it's only required at the compile time and not needed at runtime, hence its declared as provided.

**Pull request merged**